### ISSUE TITLE
Up and Down arrow keys move your view when scrolling away from selected items

### ISSUE THREAD

**Issue by danielkrajnik** [cboxdoerfer/fsearch#468]
**Is your feature request related to a problem? Please describe.**
(FSearch 0.3.alpha0 (AUR-devel)) 
When scrolling selection "stays behind" on the last selected item. It makes the up and down keys "scroll you back" to that item even when you "scrolled away" from it already (because up and down keys move the selection and not the view).  

1. some item selected
![image](https://user-images.githubusercontent.com/15096799/228850178-3f1ed3b2-f57a-4d36-96dd-cf255d78e15e.png)

2. "scroll away"
![image](https://user-images.githubusercontent.com/15096799/228850366-16cfb123-712b-485a-9788-645dc5107118.png)

3. press down arrow key - it "scrolls you back" to that item
![image](https://user-images.githubusercontent.com/15096799/228850556-3dee19d6-d77d-417f-8c6f-c4a206160017.png)

**Describe the solution you'd like**
Three ideas:
1. Selection travels with you when pressing Page Up and Page Down keys, but not when scrolling with the mouse (I think voidtools' Everything works this way).  
2. Option to make up and down arrow keys to scroll rather than move selection (or be able to assign another key for that purpose).
3. New button to put selection in the middle of the current view.

**Comment by cboxdoerfer** [comment]
I think you basically want to have a different behavior for the cursor within the list. The cursor is basically the keyboard focus within the list and it is highlighted by a rectangle around the row. So when you do any keyboard navigation (up/down, page-up/down) you're actually moving the cursor up and down and if it is currently out of view the list scrolls to the new cursor position.

> Selection travels with you when pressing Page Up and Page Down keys, but not when scrolling with the mouse (I think voidtools' Everything works this way).

Do you mean the cursor should also move up and down when you scroll with the mouse? I'm not sure I like that, but I'll see how this works and feels in Everything and report back to you.

> Option to make up and down arrow keys to scroll rather than move selection (or be able to assign another key for that purpose).

I like this one better; having a dedicated shortcut for scrolling the list with the keyboard, without moving the cursor. It would be great to know if there are other GUI apps out there which support that, for consistency sake. To me the VIM-like shortcuts sound like a good candidate, something like: 

* `ctrl-j` to scroll down by one row
* `ctrl-k` to scroll up by one row
* `ctrl-h` to scroll to the left by a bit
* `ctrl-l` to scroll to the right by a bit
* `ctrl-d` to scroll down by half a half a page
* `ctrl-u` to scroll up by half a page

But this probably feels pretty weird to non-VIM users.

Another solution I can think of would be to use the `Alt` modifier for that, so `Alt-Up`, `Alt-Down`, `Alt-PageUp`, ... would then only scroll up and down.

> New button to put selection in the middle of the current view.

That also sounds like a good action to have. I'm not sure if it makes sense to assign a default shortcut for that (because I don't know of any app which supports this by default), but with the future custom-keyboard shortcuts you could then simply assign any key-combo you want to that action.

**Comment by danielkrajnik** [comment]
> having a dedicated shortcut for scrolling the list with the keyboard, without moving the cursor

yes, that would be great. Emacs windows have this behaviour (Doom emacs at least), but it might be strange for the uninitiated.

> Do you mean the cursor should also move up and down when you scroll with the mouse

Sorry, I meant that it would move when scrolling with "Page Up" and "Page Down", but wouldn't move when scrolling with the mouse. 
I don't have Everything running right now, but I think when you scrolled and pressed either "Up" or "Down" arrow keys the cursor would appear in the top or bottom of the VISIBLE entries. The cursor wouldn't travel when scrolling, but would only appear when pressing one of those arrow keys.
